packages/grz-pydantic-models/src/grz_pydantic_models/submission/metadata/v1.py

```diff
@@ -842,6 +844,90 @@ def ensure_unpaired_long_read_data(self) -> Self:
             raise ValueError("Long read libraries can't be paired-end.")
         return self
 
+    @model_validator(mode="after")
+    def validate_sequencing_file_exists(self):  # noqa: C901, PLR0912
+        """
+        Check if there is a FASTQ or BAM file (depending on library type)
+        """
+        if self.sequence_data is None:
+            # Skip if no sequence data is present
+            return self
+
+        fastq_files = self.sequence_data.list_files(FileType.fastq)
+        bam_files = self.sequence_data.list_files(FileType.bam)
+
+        if self.library_type.endswith("_lr"):
+            if len(fastq_files) + len(bam_files) == 0:
+                raise ValueError("Long-read datasets must contain at least one BAM or FASTQ file!")
+        else:
+            if len(bam_files) > 0:
+                raise ValueError("Short-read datasets cannot contain BAM files!")
+            if len(fastq_files) == 0:
+                raise ValueError("Short-read datasets must contain at least one FASTQ file!")
+
+        if self.sequencing_layout == SequencingLayout.paired_end:
+            # check if read order is specified
+            for i in fastq_files:
+                if i.read_order is None:
+                    raise ValueError(
+                        f"Error in lab datum '{self.lab_data_name}': "
+                        f"No read order specified for FASTQ file '{i.file_path}'!"
+                    )
+
+            key = lambda f: (f.flowcell_id, f.lane_id)
+            fastq_files.sort(key=key)
+            for _key, group in groupby(fastq_files, key):
+                flowcell_id = _key[0]
+                lane_id = _key[1]
```

Comment:
can we destructure key in the loop header?
```suggestion
            for (flowcell_id, lane_id), group in groupby(fastq_files, key):
```